### PR TITLE
Lesson 04: Adding exercises and assignments

### PR DESCRIPTION
**Pull request by mitch334**

### CODE REVIEW

File: students/mitch334/lesson04/mailroom.py [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+import datetime
+
+''' Initial donor list with donation history
+    Data: Donor Name: [Donation Amount(s)]'''
+donor_db = {"William Gates, III": [100.00, 50.00],
+            "Jeff Bezos": [1000.00, 10.00, 500.00],
+            "Mark Zuckerberg": [200.00, 20.00, 50.00],
+            "Warren Buffet": [600.00, 300.00],
+            "Paul Allen": [1000.00]
+            }
```
Comment (natasha-aleksandrova) [inline students/mitch334/lesson04/mailroom.py]: 👍

File: students/mitch334/lesson04/mailroom.py [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+# Finally, use string formatting to compose an email thanking the donor for their generous donation. Print the email to the terminal and return to the original prompt.
+# It is fine (for now) for the program not to store the names of the new donors that had been added, in other words, to forget new donors once the script quits running.
+def get_donor_names():
+    donor_names = []
+    for donor in donor_db:
+        donor_names.append(donor)
+
+    return donor_names
+
+def create_donation(name):
```
Comment (natasha-aleksandrova) [inline students/mitch334/lesson04/mailroom.py]: small style violation: 2 spaces for module functions.

File: students/mitch334/lesson04/mailroom.py [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+    donor_names = get_donor_names()
```
Comment (natasha-aleksandrova) [inline students/mitch334/lesson04/mailroom.py]: doesn't look like this is being used?

File: students/mitch334/lesson04/mailroom.py [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+                            '\tOptions:\n'
+                            "\t\tEnter 'list' for donor list\n"
+                            "\t\tEnter 'exit' to return to the main menu\n"
+                            '\tEnter full name of donor: ')
+
+        donor_names = get_donor_names()
+
+        if donor_name.lower() == 'list':
+            print('\nDonor List:')
+            for donor in donor_db: print('\t',f'{donor}')
```
Comment (natasha-aleksandrova) [inline students/mitch334/lesson04/mailroom.py]: this should be two separate lines

File: students/mitch334/lesson04/mailroom.py [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+
+        elif donor_name.lower() == 'db':
+            print('\nDonor Database:')
+            for name, val in donor_db.items(): print('\t',f'{name:26}{val}')
```
Comment (natasha-aleksandrova) [inline students/mitch334/lesson04/mailroom.py]: same here, two separate lines

File: students/mitch334/lesson04/mailroom.py (hunk 3 above)
Comment (mitch334) [reply students/mitch334/lesson04/mailroom.py]: Updated.

File: students/mitch334/lesson04/mailroom.py (hunk 4 above)
Comment (mitch334) [reply students/mitch334/lesson04/mailroom.py]: Removed.

File: students/mitch334/lesson04/mailroom.py (hunk 3 above)
Comment (mitch334) [reply students/mitch334/lesson04/mailroom.py]: Removed.

File: students/mitch334/lesson04/mailroom.py (hunk 4 above)
Comment (mitch334) [reply students/mitch334/lesson04/mailroom.py]: Updated.

File: students/mitch334/lesson04/mailroom.py (hunk 5 above)
Comment (mitch334) [reply students/mitch334/lesson04/mailroom.py]: Updated.

**Pull request merged**